designcompose/src/main/java/com/android/designcompose/DesignFrame.kt

```diff
@@ -84,6 +84,60 @@ internal fun DesignFrame(
         m = m.then(customModifier)
     }
 
+    // Keep track of the layout state, which changes whenever this view's layout changes
+    val (layoutState, setLayoutState) = remember { mutableStateOf(0) }
+    var rootLayoutId = parentLayout?.rootLayoutId ?: -1
+    if (rootLayoutId == -1) rootLayoutId = layoutId
+
+    val subscribeLayout =
```

Comment:
I like that this logic is getting better encapsulated, tho I think we need to revisit the implementation soon (squoosh is one alternative since only the root runs layout, I was also looking at the `Modifier.Node` API which allows traversal of children -- that might give a way to gather all of the child layout properties after composition but before layout, but I'm not sure...).

Reply:
Interesting I haven't seen `Modiifer.Node`, worth checking out.